File: day01/main.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"strconv"
)

func main() {
	var numbers []int

	src, err := os.Open("./input.txt")
	if err != nil {
		log.Fatalf("failed to read file into scanner: %v", err)
	}
	defer src.Close()

	scanner := bufio.NewScanner(src)
	for scanner.Scan() {
		line := scanner.Text()
		freqCh, err := strconv.Atoi(line)
		if err != nil {
			log.Fatalf("Failed to parse line into numbber: %v", err)
		}
		numbers = append(numbers, freqCh)
	}

	sol1 := prob1(numbers)
	fmt.Printf("Solution for prob1 is: %d\n", sol1)
	sol2 := prob2(numbers)
	fmt.Printf("Solution for prob2 is: %d\n", sol2)

}

func prob1(numbers []int) (freq int) {
	for _, v := range numbers {
		freq += v
	}
	return
}

func prob2(numbers []int) (freq int) {
	freqL := make(map[int]int)

	ok := true
	for ok {
		for _, v := range numbers {
			freq += v
			freqL[freq]++
			if freqL[freq] >= 2 {
				ok = false
				break
			}
		}
	}
	return
}

File: day01/src/main.rs
use std::collections::HashMap;
use std::fs::File;
use std::io::{BufRead, BufReader};

fn main() {
    let file = File::open("input.txt").expect("Failed to open file");

    let mut _numbers: Vec<i32> = vec![];
    for line in BufReader::new(file).lines() {
        _numbers.push(
            line.expect("Failed to read the line")
                .parse()
                .expect("Failed to parse into i32"),
        );
    }

    let mut freq = 0;
    let mut freq_list: HashMap<i32, i32> = HashMap::new();
    let mut ok: bool = true;
    let mut first: bool = true;
    while ok {
        for v in &_numbers {
            freq += v;
            let c = freq_list.entry(freq).or_insert(1);
            if c >= &mut 2 {
                ok = false;
                break;
            }
            *c += 1;
        }
        if first {
            println!("Solution for problem 1 is {}", freq);
            first = false;
        }
    }
    println!("Solution for problem 2 is {}", freq);
}
